otter/generate/autograder.py

```diff
@@ -128,12 +130,33 @@ def main(tests_path, output_path, config, lang, requirements, overwrite_requirem
         template_context["other_requirements"] = f.read()
         template_context["overwrite_requirements"] = overwrite_requirements
 
+        # close the {% if not other_requirements %}stream
+        f.close()
+        
+        # open environment if it exists
+        # unlike requirements.txt, we will always overwrite, not append by default
+        environment = environment
+        env_filename = "environment.yml"
+        if environment is None and os.path.isfile(env_filename):
+            environment = env_filename
+        
+        if environment:
+            assert os.path.isfile(environment), f"Environment file {environment} not found"
+            with open(environment) as f:
+                data = yaml.safe_load(f)
+                f.close()
+            if "name" not in data:
```

Comment:
we should **always** overwrite name, not just when it is not present (otherwise otter will not be able to access the environment)